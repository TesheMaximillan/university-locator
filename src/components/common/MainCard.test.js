import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Card from './_mocks_/MainCard';
import renderContext from '../../redux/storeMock';

describe('MainCard component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<BrowserRouter><Card /></BrowserRouter>));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });
});
