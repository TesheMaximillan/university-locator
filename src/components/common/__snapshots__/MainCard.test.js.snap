// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard component should render with given state from redux store 1`] = `
<a
  className="cardContainer"
  href="/details/1"
  onClick={[Function]}
>
  <svg
    className="icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M262.62 336L342 256l-79.38-80m68.35 80H170"
      fill="none"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="32"
    />
    <path
      d="M256 448c106 0 192-86 192-192S362 64 256 64 64 150 64 256s86 192 192 192z"
      fill="none"
      strokeMiterlimit="10"
      strokeWidth="32"
    />
  </svg>
  <svg
    baseProfile="tiny"
    className="location"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    version="1.2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.657 5.304c-3.124-3.073-8.189-3.073-11.313 0-3.124 3.074-3.124 8.057 0 11.13l5.656 5.565 5.657-5.565c3.124-3.073 3.124-8.056 0-11.13zm-5.657 8.195c-.668 0-1.295-.26-1.768-.732-.975-.975-.975-2.561 0-3.536.472-.472 1.1-.732 1.768-.732s1.296.26 1.768.732c.975.975.975 2.562 0 3.536-.472.472-1.1.732-1.768.732z"
    />
  </svg>
  <div
    className="unvName"
  >
    <h3 />
    <p>
      chatperId
    </p>
  </div>
</a>
`;
